pdal/KDIndex.hpp

```diff
@@ -260,25 +262,29 @@ class PDAL_DLL KD3Index : public KDIndex<3>
         pt.push_back(y);
         pt.push_back(z);
         m_index->findNeighbors(resultSet, &pt[0], nanoflann::SearchParams());
+        std::valarray<PointId> temp(output.data(), output.size());
```

Comment:
This seems like a lot of copying for very little benefit.  And it made me look up what a valarray is.

if (stride > 1)
{
   for (i = 1; i < k; i++)
      output[i] = output[i * stride]
   output.resize(k);
}

Also, you should make it clear in a comment that you're still getting 'k' neighbors, but you're really extracting k * stride neighbors and then taking every stride'th item.

Reply:
Right, so that'll work. Probably went too far out of my way to get slice to work. Wish I could simply use it directly on the vector.